core/templates/services/math-interactions.service.spec.ts

```diff
@@ -205,6 +214,77 @@ describe('MathInteractionsService', () => {
     expect(mathInteractionsService.getWarningText()).toBe(
       'Your answer contains an invalid term: 3.4.5');
 
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a4', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer contains invalid variable multiplication: a4. ' +
+      'The variable should come after the number: 4a.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'a * 4', ['a'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your answer contains invalid variable multiplication: a*4. ' +
+      'The variable should come after the number: 4*a.');
+
+    expect(mathInteractionsService.validateAlgebraicExpression(
+      'x^y^z', ['x', 'y', 'z'])).toBeFalse();
+    expect(mathInteractionsService.getWarningText()).toBe(
+      'Your expression contains an exponent in an exponent.');
```

Comment:
Message seems incomplete -- do we need something like "which is not supported"? /cc @srushtirk

Ditto below.

Reply:
Done.